earthaccess/__init__.py

```diff
@@ -36,7 +37,30 @@
     "Store",
 ]
 
-__auth__ = Auth()
-__store__: Any = None
-
 __version__ = version("earthaccess")
+
+_auth = Auth()
+_store = None
+_lock = threading.Lock()
+
+
+def __getattr__(name):  # type: ignore
```

Comment:
I like this. I think we could use a docstring.

Reply:
Added 👍

Reply:
I forget when module-level getattrs were added to Python, but I vaguely feel like it wasn't that long ago. My sense of time is often a complete mess :) I think the answer is "no", but just want to double-check: are there any Python versions (that are not EOL) that don't support this?

Reply:
No worries. It was added in Python 3.7 https://peps.python.org/pep-0562 so we should be good